src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxButton.java

```diff
@@ -65,6 +86,12 @@ public void setArmed(final boolean armed) {
         });
 
         setEnabled(comboBox.isEnabled());
+
+        AccessibleContext ac = this.getAccessibleContext();
+        if (ac != null && !comboBox.isEditable() && list.getModel().getSize() > 0) {
+            ac.setAccessibleName(String.valueOf(list.getModel()
+                    .getElementAt(0)));
+        }
```

Comment:
So i see it updates accessible name whenever selection is changed. What about the initial value? Will the combobox initial value be displayed if it was never changed?

Reply:
> What about the initial value? Will the combobox initial value be displayed if it was never changed?

The initial value is set in `AquaComboBoxButton` constructor. Yes, initial value is displayed with screen magnifier even if it has never changed.